backends/fs/fs.go

```diff
@@ -22,6 +22,9 @@ type Backend struct {
 
 func (b *Backend) List(ctx context.Context, prefix string) (simpleblob.BlobList, error) {
 	var blobs simpleblob.BlobList
+	if err := ctx.Err(); err != nil {
+		return blobs, err
+	}
 
 	entries, err := os.ReadDir(b.rootPath)
```

Comment:
Might be overkill here but it seems it is generally recommended to check for context cancellation at regular intervals if processing a large amount of entries. Maybe a potential improvement?

In the VT Feed (which admittedly has hundreds of millions of records), we do it like this:
https://github.com/PowerDNS/platform-filter/blob/vtfeed/go/oxfeed/vtfeed/feed.go#L122